File: application/src/main/java/de/pbauerochse/worklogviewer/youtrack/issuedetails/IssueDetailsResponse.java
package de.pbauerochse.worklogviewer.youtrack.issuedetails;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

/**
 * Created by patrick on 31.10.15.
 * {"issue":[{"id":"PATRICK-1","entityId":"87-2","jiraId":null,"field":[],"comment":[],"tag":[]},{"id":"PATRICK-2","entityId":"87-4","jiraId":null,"field":[{"name":"resolved","value":"1446292986266"}],"comment":[],"tag":[]}]}
 */
@Deprecated
@JsonIgnoreProperties(ignoreUnknown = true)
public class IssueDetailsResponse {

    private final List<IssueDetails> issues;

    @JsonCreator
    public IssueDetailsResponse(@JsonProperty("issue") List<IssueDetails> issues) {
        this.issues = issues;
    }

    public List<IssueDetails> getIssues() {
        return issues;
    }
}

File: connectors/v20174_to_20181/src/main/java/de/pbauerochse/worklogviewer/connector/v2017/domain/issuedetails/IssueDetailsResponse.kt
package de.pbauerochse.worklogviewer.connector.v2017.domain.issuedetails

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

/**
 * The YouTrack API response object
 * when fetching the details for a
 * bunch if Issues
 */
@JsonIgnoreProperties(ignoreUnknown = true)
data class IssueDetailsResponse @JsonCreator constructor(
    @JsonProperty("issue") val issues : List<IssueDetails>
)